R/wwinference.R

```diff
@@ -68,7 +70,17 @@ wwinference <- function(ww_data,
                         ),
                         mcmc_options = wwinference::get_mcmc_options(),
                         generate_initial_values = TRUE,
-                        compiled_model = wwinference::compile_model()) {
+                        compiled_model = wwinference::compile_model(),
+                        dist_matrix = as.matrix(
```

Comment:
I would recommend not having a default for this. Instead, I would pass it in right underneath the `ww_data` and `hosp_data` as a required argument to the function. Otherwise, the user could use this function without realizing they are supplying this distance matrix as input data.

Reply:
This will require you to add it to the vignette though and pass it to `wwinference(dist_matrix = etc)`.

For this PR, I would recommend just hard coding it into the vignette with the values you used to generate the spatially correlated data. In a separate issue, I would recommend adding the distance matrix used to generate the spatially correlated data as package data. That way, if you change it, it will be updated in the `wwinference.Rmd`

Reply:
> I would recommend not having a default for this. Instead, I would pass it in right underneath the `ww_data` and `hosp_data` as a required argument to the function. Otherwise, the user could use this function without realizing they are supplying this distance matrix as input data.

Agree, see https://github.com/CDCgov/ww-inference-model/pull/56#discussion_r1722360146

Reply:
Would strongly suggest default `NULL`, especially since we're also interested in cases where we infer a correlation structure but do not have a known distance matrix.